crates/resolver-tests/src/sat.rs

```diff
@@ -86,16 +87,17 @@ fn create_dependencies_vars<'a>(
             .or_default()
             .insert((kind, platform), solver.new_var());
 
-        let dep_feature_var_map = dep
-            .features()
-            .iter()
-            .map(|&f| (f, solver.new_var()))
-            .collect();
-
-        var_for_is_dependencies_features_used
+        let dep_feature_vars = var_for_is_dependencies_features_used
             .entry(name)
             .or_default()
-            .insert((kind, platform), dep_feature_var_map);
+            .entry((kind, platform))
+            .or_default();
+
+        for &feature_name in dep.features() {
+            if let Entry::Vacant(entry) = dep_feature_vars.entry(feature_name) {
+                entry.insert(solver.new_var());
```

Comment:
Why the change?  Is this just to avoid  allocating  several `new_var`'s? Do we really end up with dependencies that  activate the same feature more than once?

Also would this be clearer as `.entry(feature_name).or_insert_with(|| solver.new_var())`?

Reply:
> Why the change? Is this just to avoid allocating several `new_var`'s?

Yes
 
> Also would this be clearer as `.entry(feature_name).or_insert_with(|| solver.new_var())`?

I didn't use `or_insert_with` because I don't need the returned `Entry`.

Reply:
Okay, that makes sense.

> Do we really end up with dependencies that activate the same feature more than once?

For testing purposes I added a `else {todo!()}` to each of these, and all tests pass. So I don't think our test suite hits this case. What am I missing? Do you have plans that will add test code that hits this case? Are there significant downsides to a few calls to `solver.new_var()`? I'm not objecting, I'm just trying to understand.

Reply:
The `else {todo!()}` will trigger if we have a dependency with a list of features containing duplicates, or if the feature `"dep/feature"` or `"dep?/feature"` is listed multiple times among all features declared in a summary. I will add a test for this.

Reply:
> Are there significant downsides to a few calls to solver.new_var()?

Maybe not, but removing duplicates will improve the debugging if we want to manually inspect the clauses.